### DISCUSSION TITLE
Ruby zaman nesneleri

### DISCUSSION THREAD

**Discussion by BunyaminAkdeniz** [19/classroom#53] [Ruby] [answered]
Merhaba Hocam. Classımda saniyede bir çalışacak bir metot yazmak istiyorum. Bunun için aklıma şöyle bir çözüm geldi 
``` ruby
 def every_so_many_seconds(seconds)
        last_tick = Time.now
        loop do
          sleep 0.1
            if Time.now - last_tick >= seconds
              last_tick += seconds
                yield
            end
        end
    end
```

Ancak matematiksel hesap ve atama işlemleri yüzünden bazen gecikme yapıyor. Sağlıklı bir çözüm olmadığı belli. Bunu nasıl iyileştirebilirim. Teşekkürler

**Comment by roktas** [comment, marked as answer]
Bu ilginç bir soru...  İdeal durumu ifade etmem gerekirse, bildiğin/kullandığın anlamda işletim sistemleri (preemptive OS) böyle yüksek hassasiyette zamanlanmış işleri çalıştırmak için kurgulanmamıştır.  Bunun için [RTOS](https://en.wikipedia.org/wiki/Real-time_operating_system) dediğimiz gerçek zamanlı işletim sistemleri var.  Olağan işletim sistemleriyle dikkatli bir planlama ile ancak yeterli bir yaklaşıklık sağlayabilirsin.  Alıntıladığın kod da buna bir örnek (sanırım Stackoverflow'tan).  Dikkat etmen gereken hususlardan biri "saniyede bir çalışmasını istiyorum" dediğin işlemi optimize etmek ve bu işlem ne kadar sürüyorsa o süreyi periyottan çıkarmak.  Bunun ötesinde şu an önerebileceğim, aklıma gelen bir şey yok.

**Reply by BunyaminAkdeniz** [reply]
Teşekkür ederim
